tools/trace_analyzer.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright(c) 2023 Intel Corporation. All rights reserved.
+
+'''
+A tool to analyze SOF trace file.
```

Comment:
Please this a little bit longer

Input
- Trace from which tool? (we have too many). If mtrace then rename to `mtrace_perf_analyzer.py` or something more specific.
- What kind of data does it analyze?
- Over which period of time / what kind of input? Does it support streaming or just whole file? Maybe streaming in the future? (I saw a `yield`)

Output
- what kind of analysis does this do? Sliding window average? etc.
- There are some hardcoded CLK and other values: they deserve to be mentioned in the --help, at least briefly.